common/djangoapps/student/helpers.py

```diff
@@ -1,28 +1,51 @@
 """Helpers for the student app. """
 import time
 from datetime import datetime
+import json
+import urllib
+
 from pytz import UTC
+
 from django.utils.http import cookie_date
 from django.conf import settings
 from django.core.urlresolvers import reverse, NoReverseMatch
+
 import third_party_auth
-import urllib
 from verify_student.models import SoftwareSecurePhotoVerification  # pylint: disable=F0401
 from course_modes.models import CourseMode
 
 
-def set_logged_in_cookie(request, response):
-    """Set a cookie indicating that the user is logged in.
+def set_logged_in_cookies(request, response, user=None):
+    """
+    Set cookies indicating that the user is logged in.
 
     Some installations have an external marketing site configured
     that displays a different UI when the user is logged in
     (e.g. a link to the student dashboard instead of to the login page)
 
+    Currently, two cookies are set:
+
+    * EDXMKTG_LOGGED_IN_COOKIE_NAME: Set to 'true' if the user is logged in.
+    * EDXMKTG_USER_INFO_COOKIE_VERSION: JSON-encoded dictionary with user information (see below).
+
+    The user info cookie has the following format:
+    {
+        "version": 1,
+        "header_urls": {
+            "account_settings": "https://example.com/account/settings",
+            "learner_profile": "https://example.com/u/test-user",
+            "logout": "https://example.com/logout"
+        }
+    }
```

Comment:
@AlasdairSwan This is the format of the cookie I'm setting.  Does that give you what you need for the marketing site dropdown?